msg_truncate.go

```diff
@@ -71,9 +71,12 @@ func (dns *Msg) Truncate(size int) {
 		l, numExtra = truncateLoop(dns.Extra, size, l, compression)
 	}
 
-	// According to RFC 2181, the TC bit should only be set if not all
-	// of the answer RRs can be included in the response.
-	dns.Truncated = len(dns.Answer) > numAnswer
+	// According to RFC 2181, the TC bit should only be set if not all of
```

Comment:
This text can be elevated to the function doc. That also still says the TC bit is only set when the answer section is truncated.